### PR TITLE
Fix links for author detail pages and add Mastodon

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Currently author detail pages do not render links maintained in authorsData.json as they are not passed through to the collection that renders those pages.

**Before**
image

**After**
image

Additionally this PR adds the possibility to add Mastodon links for authors.

### CODE REVIEW

**Review by devnook — APPROVED** [review]

**Review by devnook — CHANGES_REQUESTED** [review]
ah sorry - deleted twitter icon 24x24 made that the twitter link from homepage footer disappeared - ptal

File: site/_data/authorsData.json [hunk 1]
```diff
@@ -803,6 +810,7 @@
     "twitter": "tunetheweb",
     "github": "tunetheweb",
     "homepage": "https://www.tunetheweb.com",
+    "mastodon": "https://mastodon.online/@tunetheweb",
```
Comment (tunetheweb) [inline site/_data/authorsData.json:813]: I moved:

```suggestion
    "mastodon": "https://webperf.social/@tunetheweb",
```

**Comment by matthiasrohmer** [comment]
@devnook, thanks for spotting! That was why it was so weirdly sized compared to the other social icons... fixed by adding the explicit width and height back in for Twitter.

**Review by devnook — APPROVED** [review]

**Comment by Aizmansor** [comment]
Dihantar daripada telefon saya-------- Mesej asal --------Daripada: Matthias Rohmer ***@***.***>Tarikh: Isn, 21 Nov 2022, 6:16 PTGKepada: "GoogleChrome/developer.chrome.com" ***@***.***>Sk: Faizul mansor ***@***.***>, Manual ***@***.***>Tajuk: Re: [GoogleChrome/developer.chrome.com] Fix links for author detail pages and add Mastodon (PR #4336)
Merged #4336 into main.

**Pull request merged**